### DISCUSSION TITLE
`@@map` error at docker postgres

### DISCUSSION THREAD

**Discussion by dwissaaj** [prisma/prisma#16631] [Q&A]
### Bug description

I have a database consist User and Bookmark and tried to change it using `@@map("users")` but throw and error, before add map function it worked fine in my postgres docker.  I try restart database but not solving. I think it because it didnt recognize the changes but when we We log prisma it actually show the difference after add map

**Log the changes**
```
$ npx prisma -v  
npx prisma db pull
Prisma schema loaded from prisma\schema.prisma
Environment variables loaded from .env
Datasource "db": PostgreSQL database "nest", schema "public" at "localhost:5434"

✔ Introspected 2 models and wrote them into prisma\schema.prisma in 126ms

*** WARNING ***

These models were enriched with `@@map` information taken from the previous Prisma schema.
- Model "Bookmark"
- Model "User"

Run prisma generate to generate Prisma Client.
```

**Error Code**

```
[Nest] 18464  - 12/05/2022, 9:49:15 AM   ERROR [ExceptionsHandler] 
Invalid `this.prisma.user.create()` invocation in
C:\Users\dwisb\WebstormProjects\expressApp\src\auth\auth.service.ts:10:45

   7 constructor(private prisma: PrismaService) {}
   8 async signIn(dto: AuthDto) {
   9   const hash = await argon.hash(dto.password);
→ 10   const userData = await this.prisma.user.create(
The table `public.User` does not exist in the current database.
Error: 
Invalid `this.prisma.user.create()` invocation in
C:\Users\dwisb\WebstormProjects\expressApp\src\auth\auth.service.ts:10:45

   7 constructor(private prisma: PrismaService) {}
   8 async signIn(dto: AuthDto) {
   9   const hash = await argon.hash(dto.password);
→ 10   const userData = await this.prisma.user.create(
The table `public.User` does not exist in the current database.
    at RequestHandler.handleRequestError (C:\Users\dwisb\WebstormProjects\expressApp\node_modules\@prisma\client
\runtime\index.js:34869:13)
    at RequestHandler.handleAndLogRequestError (C:\Users\dwisb\WebstormProjects\expressApp\node_modules\@prisma\
client\runtime\index.js:34841:12)
    at RequestHandler.request (C:\Users\dwisb\WebstormProjects\expressApp\node_modules\@prisma\client\runtime\in
dex.js:34836:12)
    at PrismaService._request (C:\Users\dwisb\WebstormProjects\expressApp\node_modules\@prisma\client\runtime\in
dex.js:35926:16)
    at AuthService.signIn (C:\Users\dwisb\WebstormProjects\expressApp\src\auth\auth.service.ts:10:22)
    at C:\Users\dwisb\WebstormProjects\expressApp\node_modules\@nestjs\core\router\router-execution-context.js:4
6:28
    at C:\Users\dwisb\WebstormProjects\expressApp\node_modules\@nestjs\core\router\router-proxy.js:9:17

```

### Reproduce

1. Tried to post request using insomnia
2. Working fine
3. Add @@map
4. error

### Expected behavior

Success 201 HTTP Request

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "./prisma/client"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id        Int        @id @default(autoincrement())
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
  email     String     @unique
  hash      String
  firstName String?
  lastName  String?
  bookmarks Bookmark[]

  @@map("users")
}

model Bookmark {
  id          Int      @id @default(autoincrement())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  title       String
  description String?
  link        String
  userId      Int
  user        User     @relation(fields: [userId], references: [id])

  @@map("bookmarks")
}

```

### Environment & setup

- OS: <! Windows-->
- Database: 
- Node.js version: 
-  Docker 

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.7.1
@prisma/client          : 4.7.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules\@prisma\engi
nes\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules\@prisma
\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules\@prisma\e
ngines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules\@prisma\engines\p
risma-fmt-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.7.1-1.272861e07ab64f234d3ffc4094e32bd61775599c
Default Engines Hash    : 272861e07ab64f234d3ffc4094e32bd61775599c
Studio                  : 0.477.0

```

**Comment by dwissaaj** [comment]
Probably relate to #16616

**Comment by janpio** [comment]
When you add `@@map`, you are telling Prisma that the data is not in the table `User` but table `user` now. Did you rename the table accordingly manually or migrate your database so that the table exists? Otherwise this error is 100% expected.

**Reply by vladi-stevanovic** [reply]
👋 Hi @dwissaaj , did you perhaps have a chance to check Jan's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by Zuhdi24** [comment]
Are you trying to follow [this tutorial](https://youtu.be/GHTA143_b-s)?
If so, i have the same problem, table users dosent exist.
My solution is to generate new prisma client by typing
```
npx prisma generate
```

**Reply by codecret** [reply]
`Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Error: 
EPERM: operation not permitted, unlink 'C:\Users\malek\Desktop\firstnestjsproject\car_manager_first\node_modules\.prisma\client\query_engine-windows.dll.node'`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
